common/rfb/CSecurityTLS.cxx

```diff
@@ -203,21 +203,46 @@ bool CSecurityTLS::processMsg()
 void CSecurityTLS::setParam()
 {
   static const char kx_anon_priority[] = ":+ANON-ECDH:+ANON-DH";
+#if GNUTLS_VERSION_NUMBER < 0x030603
+  static const char kx_gnutls_priority_fallback[] = "NORMAL";
+#endif
 
   int ret;
   char *prio;
   const char *err;
 
-  prio = (char*)malloc(strlen(Security::GnuTLSPriority) +
-                       strlen(kx_anon_priority) + 1);
+  if (strcmp(Security::GnuTLSPriority, "") != 0)
+    prio = (char*)malloc(strlen(Security::GnuTLSPriority) +
+                         strlen(kx_anon_priority) + 1);
+  else
+#if GNUTLS_VERSION_NUMBER >= 0x030603
+    prio = (char*)malloc(strlen(kx_anon_priority) + 1);
+#else
+    prio = (char*)malloc(strlen(kx_gnutls_priority_fallback) +
+                         strlen(kx_anon_priority) + 1);
+#endif
+
   if (prio == NULL)
     throw AuthFailureException("Not enough memory for GnuTLS priority string");
 
-  strcpy(prio, Security::GnuTLSPriority);
+  if (strcmp(Security::GnuTLSPriority, "") != 0)
+    strcpy(prio, Security::GnuTLSPriority);
+#if GNUTLS_VERSION_NUMBER < 0x030603
+  else
+    strcpy(prio, Security::kx_gnutls_priority_fallback);
+#endif
+
   if (anon)
     strcat(prio, kx_anon_priority);
 
+#if GNUTLS_VERSION_NUMBER >= 0x030603
+  if (strcmp(Security::GnuTLSPriority, "") == 0)
+    ret = gnutls_set_default_priority_append(session, prio, &err, 0);
+  else
+    ret = gnutls_priority_set_direct(session, prio, &err);
+#else
   ret = gnutls_priority_set_direct(session, prio, &err);
```

Comment:
Same thing here.

Reply:
I think this one is correct, but what is not correct is combination of default prio and new GnuTLS, where `prio` will be only initialized when `anon` is true.